src/Microsoft.Identity.Client/Cache/Items/MsalIdTokenCacheItem.cs

```diff
@@ -68,6 +71,7 @@ internal MsalIdTokenCacheItem()
             Secret = secret;
             RawClientInfo = rawClientInfo;
 
+            HomeAccountId = userId;
```

Comment:
>HomeAccountId = userId; [](start = 12, length = 23)

not sure about this

Reply:
What does the HomeAccountId represent in AAD? This was one of the changes I made a few months back...from what I recall the HomeAccountId needed to be set to avoid alot of exceptions from being thrown

Reply:
If the homeaccountid is defined as uid.tid, this would be correct. Adfs has no notion of tenants

Reply:
as I mention for the AT case, I really would like this behavior hidden behind a factory method or similar which makes it clear that this is for ADFS specific scenarios and also which makes it clear that for ADFS this value is not expected to exist. HomeAccountId is important for many other scenarios. Also would be great with a confirmation that having ADFS items in the cache with non- adfs items will still work. @trwalke, can you pls. help create the corresponding tests?

---
In reply to: [252387992](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/834#discussion_r252387992) [](ancestors = 252387992)